### PR TITLE
Adds SMS auth endpoints

### PR DESCRIPTION
**Pull request by tanatorn**
Adds the API endpoints required to actually do SMS auth, there will be a PR on the bank repo to actually do the hooking up of bank.hackclub.com to this.

Notable things:
* Adds a phone_number field to the Users object as well as a way to update it. Bank will eventually have to make API calls to add a phone number + enroll in SMS login
* We're going to be using Twilio's [Verify API](https://www.twilio.com/docs/verify/api) rather than storing our own login codes. This is better because:
  1. We don't actually have to purchase/manage phone numbers 
  2. They have auto expiry/rate limits and all other good things out of the box

image
image

### CODE REVIEW

File: app/services/twilio_verification_service.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'twilio-ruby'
+
+class TwilioVerificationService
+  CLIENT = Twilio::REST::Client.new(
+    Rails.application.secrets.twilio_account_sid,
+    Rails.application.secrets.twilio_api_key
+  )
+  VERIFY_SERVICE_ID = 'VAa06a66dad4c1ca3c199a46334ff11945'
```
Comment (tanatorn) [inline app/services/twilio_verification_service.rb:10]: this isn't private so it's okay to just hardcode it here

Comment (maxwofford) [reply app/services/twilio_verification_service.rb:10]: Could you add a comment above that line that so someone doesn't put it in secrets later?

**Comment by maxwofford** [comment]
@tanatorn This is looking great. When I look at [this recording](https://hackclub.slack.com/archives/C026RKHLPNJ/p1636836087015000), the code feels hard to read in a push-notification. Is there a way to shorten the code or add a hyphen in the middle to help mentally break it down?

**Comment by maxwofford** [comment]
Also, is the bank user going to get an sms & email at the same time? Will the codes be the same?

**Comment by tanatorn** [comment]
@maxwofford, unfortunately the text message is controlled fully by Twilio, there's nothing I can do there to change that. Additionally, it'll either go to SMS or Email. However, we don't manage any state on the SMS side, e.g. we'll generate a new user token, but we won't touch the `login_codes` table.

**Comment by tanatorn** [comment]
PTAL @maxwofford

**Review by maxwofford — APPROVED** [review]
This looks great! Please add the heroku env keys before merge if you haven't yet.

I have a couple questions about user experience I'll leave on the bank PR.

**Comment by tanatorn** [comment]
done!

**Pull request merged**